src/array/broadcast.jl

```diff
@@ -67,6 +67,20 @@ function Base.copyto!(dest::AbstractDimArray, bc::Broadcasted{DimensionalStyle{S
     end
 end
 
+# This is needed for GPUs to prevent scalar indexing problems for things like
+# d .= 1:10
+function Base.copyto!(dest::AbstractDimArray, bc::Broadcasted{Nothing})
```

Comment:
```suggestion
function Base.copyto!(dest::AbstractDimArray, bc::Broadcasted)
```

Why not just this?

Reply:
Turns out we can't... its ambiguous

Reply:
Ya I had tried the exact same thing at first and got a bunch of ambiguities. It is sad :(